util/config/make_sys_basic_types.py

```diff
@@ -0,0 +1,230 @@
+    '32767': 'int(16)',
+    '65535': 'uint(16)',
+    '2147483647': 'int(32)',
+    '4294967295': 'uint(32)',
+    '9223372036854775807': 'int(64)',
+    '18446744073709551615': 'uint(64)',
+}
+
+
+def main():
+    """Parse command line arguments, create module, and print to stdout."""
+    args = _parse_args()
+    _setup_logging(args.verbose)
+    print(get_sys_c_types(args.doc))
+
+
+def get_sys_c_types(docs=False):
+    """Returns a string with the SysCTypes.chpl module content."""
+
+    # Find the $CHPL_HOME/util/config/ dir.
+    util_cfg_dir = os.path.abspath(os.path.dirname(__file__))
+    logging.debug('Util config dir: {0}'.format(util_cfg_dir))
+
+    # Get the C compile line.
+    compileline_cmd = os.path.join(util_cfg_dir, 'compileline')
+    cmd = '{compileline_cmd} --compile'.format(**locals())
+    compileline = os.popen(cmd).read().strip()
+    logging.debug('Compile line: {0}'.format(compileline))
+
+    # Create temp header file with *_MAX macros, then run it through the C
+    # preprocessor using the C compile line found above.
+    h_file = 'find_int_sizes_tmp.h'
+    with _ensure_deleted(h_file):
+        logging.debug('Creating temp header: {0}'.format(h_file))
+        with open(h_file, 'w') as fp:
+            fp.write("""
+#include "sys_basic.h"
+#include <limits.h>
+#include <stdint.h>
+#include <math.h>
+
+FIND_INT_SIZES_START
+""")
+
+            for max_macro, _, _ in _types:
+                fp.write('{0}\n'.format(max_macro))
+            logging.debug('Wrote {0} max types to {1}'.format(len(_types), h_file))
+
+        compile_cmd = '{compileline} -E {h_file}'.format(**locals())
+        logging.debug('Preprocessor command: {0}'.format(compile_cmd))
+        compile_result = os.popen(compile_cmd).read()
+        logging.debug('Captured preprocessor output with {0} '
+                      'characters.'.format(len(compile_result)))
+
+    # Iterate through the preprocessor output, find the max value expressions,
+    # and record them in a list.
+    max_exprs = []
+    keep = False
+    for line in compile_result.splitlines():
+        if line.startswith('#') or line.startswith('//'):
+            continue
+        elif 'FIND_INT_SIZES_START' in line:
+            keep = True
+        elif keep:
+            max_exprs.append(line.strip())
+    logging.debug('Found {0} lines of max type '
+                  'values.'.format(len(max_exprs)))
+
+    if len(max_exprs) != len(_types):
+        logging.error('Found {0} max values, but {1} types were '
+                      'expected.'.format(len(max_exprs), len(_types)))
+        sys.exit(1)
+
+    # Iterate through the max value expressions, evaluate each one, and store
+    # it in a list. Python deals with arbitrarily large integers, so there is
+    # no fear of overflow.
+    replace_pattern = re.compile(r'[UL]', re.IGNORECASE)
+    max_values = []
+    for expr in max_exprs:
+        ex = re.sub(replace_pattern, '', expr)
+        value = eval(ex)
```

Comment:
I tried to find a better (safer) way to do this with the ast and compiler modules, but it requires a lot more work. And we have pretty good control over the input, so I think it'll be ok... at least for now.